externals/llvm-external-projects/torch-mlir-dialects/include/torch-mlir-dialects/Dialect/Tcp/IR/TcpOps.td

```diff
@@ -514,4 +514,28 @@ def Tcp_Atan2Op : Tcp_BinaryElementwiseOp<"atan2", [SameOperandsAndResultElement
   let assemblyFormat = "$in1 `,` $in2 attr-dict `:` type($in1) `,` type($in2) `->` type($out)";
 }
 
+def Tcp_CastOp:  Tcp_Op<"cast", [Pure, Elementwise, SameOperandsAndResultShape]> {
+
+  let summary = "TCP Cast operation";
+
+  let description = [{
+    Performs elementwise type conversion.
+  }];
+
+  let arguments = (ins
+    Tcp_Tensor:$in,
+    // TODO: Better way to align with enum SignednessSemantics
+    OptionalAttr<UI32Attr>:$in_int_element_type,
```

Comment:
If this is only meant to represent the `SignednessSemantics`, we should probably name it accordingly, say something like `in_int_signedness`. 

My earlier comment was just that `dtype` is not a term used in core MLIR.